server/config/configresource/config_resource.go

```diff
@@ -54,245 +60,100 @@ func Repository(db *sql.DB, opts ...option) *repository {
 
 }
 
-const ResourceID = "/app/config/update"
+const (
+	ResourceID   = "/app/config/update"
+	ResourceName = "Config"
+)
 
 type publisher interface {
 	Publish(resourceID string, message any)
 }
 
-type repository struct {
+type Repository struct {
 	db        *sql.DB
 	publisher publisher
 }
 
-func (r *repository) publish(config Config) {
+func (r *Repository) publish(config Config) {
 	if r.publisher == nil {
 		return
 	}
 
 	r.publisher.Publish(ResourceID, config)
 }
 
-func (r *repository) Current(ctx context.Context) Config {
-	defaultConfig := Config{
-		AnalyticsEnabled: true,
-	}
-
-	list, err := r.List(ctx, 1, 0, "", "", "")
-	if err != nil || len(list) != 1 {
+func (r *Repository) Current(ctx context.Context) Config {
+	cfg, err := r.Get(ctx, id.ID("current"))
+	if err != nil {
 		// TODO: log error
 		return defaultConfig
 	}
 
-	return list[0]
-}
-
-func (r *repository) SetID(cfg Config, id id.ID) Config {
-	cfg.ID = id
 	return cfg
 }
 
-const insertQuery = `
-		INSERT INTO configs (
-			"id",
-			"name",
-			"analytics_enabled"
-		) VALUES ($1, $2, $3)`
+const selectQuery = `SELECT "analytics_enabled" FROM config`
 
-func (r *repository) Create(ctx context.Context, cfg Config) (Config, error) {
-	tx, err := r.db.BeginTx(ctx, nil)
-	if err != nil {
-		return Config{}, err
-	}
-
-	_, err = tx.ExecContext(ctx, insertQuery,
-		cfg.ID,
-		cfg.Name,
-		cfg.AnalyticsEnabled,
-	)
-
-	if err != nil {
-		tx.Rollback()
-		return Config{}, fmt.Errorf("sql exec: %w", err)
-	}
-
-	err = tx.Commit()
-	if err != nil {
-		return Config{}, fmt.Errorf("commit: %w", err)
-	}
-
-	return cfg, nil
+var defaultConfig = Config{
+	ID:               id.ID("current"),
+	Name:             "Config",
+	AnalyticsEnabled: true,
 }
 
-const updateQuery = `
-		UPDATE configs SET
-			"name" = $2,
-			"analytics_enabled" = $3
-		WHERE "id" = $1`
-
-func (r *repository) Update(ctx context.Context, updated Config) (Config, error) {
-	cfg, err := r.Get(ctx, updated.ID)
-	if err != nil {
-		return Config{}, err
-	}
-
-	tx, err := r.db.BeginTx(ctx, nil)
-	if err != nil {
-		return Config{}, err
-	}
-
-	_, err = tx.ExecContext(ctx, updateQuery,
-		cfg.ID,
-		updated.Name,
-		updated.AnalyticsEnabled,
-	)
-
-	if err != nil {
-		tx.Rollback()
-		return Config{}, fmt.Errorf("sql exec: %w", err)
-	}
-
-	err = tx.Commit()
-	if err != nil {
-		return Config{}, fmt.Errorf("commit: %w", err)
-	}
-
-	r.publish(updated)
-
-	return updated, nil
-}
-
-const getQuery = baseSelect + `WHERE "id" = $1`
-
-func (r *repository) Get(ctx context.Context, id id.ID) (Config, error) {
-	cfg := Config{}
+func (r *Repository) Get(ctx context.Context, i id.ID) (Config, error) {
+	cfg := defaultConfig
 
 	err := r.db.
-		QueryRowContext(ctx, getQuery, id).
+		QueryRowContext(ctx, selectQuery).
 		Scan(
-			&cfg.ID,
-			&cfg.Name,
 			&cfg.AnalyticsEnabled,
 		)
 
 	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return cfg, nil
```

Comment:
since `config` must always have a value, `not found` actually means, `defaults not overriden`